src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsTerm.cs

```diff
@@ -90,5 +98,31 @@ private IEdmTypeReference ComputeType()
         {
             return CsdlSemanticsModel.WrapTypeReference(this.Context, this.term.Type);
         }
+
+        private IEdmTerm ComputeBaseTerm()
+        {
+            if (this.term.BaseTermName != null)
+            {
+                IEdmTerm baseTerm = this.Context.FindTerm(this.term.BaseTermName);
+                if (baseTerm != null)
+                {
+                    // Evaluate the inductive step to detect cycles.
+                    // Overriding BaseTerm getter from concrete type implementing IEdmTerm will be invoked to
+                    // detect cycles. The object assignment is required by compiler only.
+                    IEdmTerm baseTerm2 = baseTerm.BaseTerm;
+                }
+
+                return baseTerm ?? new UnresolvedVocabularyTerm(this.term.BaseTermName);
+            }
+
+            return null;
+        }
+
+        // Resolves the real name of the base term, in case it was using an alias before.
+        protected string GetCyclicBaseTermName(string baseTermName)
+        {
+            IEdmTerm schemaBaseTerm = this.Context.FindTerm(baseTermName);
+            return (schemaBaseTerm != null) ? schemaBaseTerm.FullName() : baseTermName;

```

Comment:
super nit: I prefer the null propagation syntax here: `return schemaBaseTerm?.FullName() ?? baseTermName`